reltime/reltime.py

```diff
@@ -33,14 +33,20 @@
 RELATIVE_DMY = "(" + EXP2 + " (" + DMY + "|" + WEEK_DAY + "|" + MONTH + "))"
 RELATIVE_DAY = "(" + REL_DAY + ")"
 ISO = "(((\d{4}|\d{2}|\d{1})[/-](\d{4}|\d{2}|\d{1})[/-](\d{4}|\d{2}|\d{1})(?: \d+:\d+:\d+)?))"
-YEAR = "(((?<=\s)\d{4}|^\d{4}))[\!\,\.\s+]"  # need punct/space at end of year
-DAY_MONTH = "\s+((\d{1,2}[/-]\d{1,2}))[\!\,\.\s+]"  # need punct/space at beginning/end
+YEAR = "(((?<=\s)\d{4}|^\d{4}))([\s.,!]|$)"  # need punct/space at end of year
+DAY_MONTH = "\s+((\d{1,2}[/-]\d{1,2}))([\s.,!]|$)"  # need punct/space at beginning/end
 MONTH_DAY = '((' + MONTH + ')' + "\s\d{1,2})"
-WEEKDAY_ONLY = '\s+((every\s+)?' + WEEK_DAY + '(\s+night)?)[\!\,\.\s+]'
+WEEKDAY_ONLY = '\s+((every\s+)?' + WEEK_DAY + '(\s+night)?)([\s.,!]|$)'
 HOLIDAY_ONLY = '\s+(' + HOLIDAY + ')[\!\,\.\s+]'
 RECURRING_DAILY = '((every day|everyday|every night))'
 TODAY = '((this morning|this afternoon|this evening|last day))'
 
+# times
+AM_PM = "((\s|^)(\d{1,2})(:\d{2})?(\s)?(am?|pm?)([\s.,!]|$))"
+NO_AM_PM = "((\s|^)(as of|until|around|at|by|from)( about| around| approx(\.)?(imately)?)?(\s)(\d{1,2})(:\d{1,2})?([\s.,!]|$))"
```

Comment:
there's quite a bit of absurd capturing group nesting going on to make this all run